pkg/sass-parser/lib/src/expression/number.ts

```diff
@@ -0,0 +1,121 @@
+// Copyright 2024 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import * as postcss from 'postcss';
+
+import {LazySource} from '../lazy-source';
+import type * as sassInternal from '../sass-internal';
+import * as utils from '../utils';
+import {Expression} from '.';
+
+/**
+ * The initializer properties for {@link NumberExpression}.
+ *
+ * @category Expression
+ */
+export interface NumberExpressionProps {
+  value: number;
+  unit?: string;
+  raws?: NumberExpressionRaws;
+}
+
+/**
+ * Raws indicating how to precisely serialize a {@link NumberExpression}.
+ *
+ * @category Expression
+ */
+export interface NumberExpressionRaws {
+  /**
+   * The raw string representation of the number.
+   *
+   * Numbers can be represented with or without leading and trailing zeroes, and
+   * use scientific notation. For example, the following number representations
+   * have the same value: `1e3`, `1000`, `01000.0`.
+   */
+  raw?: string;
+
+  /**
+   * The numeric value for the raw string that represents the number.
+   *
+   * This `value` is compared with {@link NumberExpression.value}. If they
+   * match, then the {@link raw} string is output when stringifying the
+   * expression. Otherwise, the {@link raw} string is ignored, and the
+   * {@link NumberExpression.value} is output.
+   */
+  value?: number;
```

Comment:
These should be in a single object, not at the root of the raws. Once https://github.com/sass/dart-sass/pull/2389 lands, you can use `RawWithValue` for this.